src/tokenize/tokenize.ts

```diff
@@ -39,38 +40,61 @@ import { categorizeTextRow } from "./textCategorize";
 import { existsSync } from "fs";
 import { mkdir, writeFile } from "fs/promises";
 import * as cheerio from "cheerio";
+import { TypedCatalogEntry } from "../classify";
+
+export const tokenizeEntry = async (
+  entry: TypedCatalogEntry,
+): Promise<TokenizedCatalogEntry> => {
+  const tokenized = await fetchAndTokenizeHTML(entry.url);
+  return { ...entry, tokenized };
+};
 
 /**
  * Fetch html for page and convert into intermediate representation (IR)
  *
  * @param url The url of the page to tokenize
  */
-export const fetchAndTokenizeHTML = async (url: URL): Promise<HDocument> => {
+const fetchAndTokenizeHTML = async (url: URL): Promise<HDocument> => {
   const html = await wrappedGetRequest(url.href);
   const token = await tokenizeHTML(cheerio.load(html));
+
   const majorName = token.majorName;
   const year = token.yearVersion;
-  const degree = majorName.includes("Minor") ? "minor" : "major";
+  const degree = majorName.includes("Minor") ? "minors" : "majors";
+  // in the url, the college will always be at the 6th place in the url
+  const college = url.toString().split("/")[6].replaceAll("-", "_");
```

Comment:
This was a temporary solution because of how the pipeline works right now (doesn't parse the college from the filtering stage). In the next refactoring we will have most of the meta data of the majors in the `classify` stage for better convention